### PR TITLE
clean read_only_cache in the bg

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Problem
Several problems with read only cache lru cleaning.

#### Summary of Changes
Move lru cleaning to a background thread.

Fixes #

### CODE REVIEW

File: runtime/src/read_only_accounts_cache.rs [hunk 1]
```diff
@@ -27,25 +28,54 @@ pub struct ReadOnlyAccountCacheEntry {
...
     hits: AtomicU64,
     misses: AtomicU64,
-    lru: LruList,
     per_account_size: usize,
+    stop: Arc<AtomicBool>,
+}
+
+impl Drop for ReadOnlyAccountsCache {
+    fn drop(&mut self) {
+        self.stop.store(true, Ordering::Relaxed);
```
Comment (sakridge) [inline runtime/src/read_only_accounts_cache.rs:42]: generally good practice to join the thread as well

File: runtime/src/read_only_accounts_cache.rs [hunk 2]
```diff
@@ -92,86 +140,86 @@ impl ReadOnlyAccountsCache {
...
-            list.sort();
-            new_size = self.purge_lru_list(&mut list, false, new_size);
-            // the list is stored in self so we use it to purge next time
+            // purge from the lru list we last made
+            if self.purge_lru_list(&lru, &mut lru_index) {
+                continue;
+            }
+
+            // we didn't get enough, so calculate a new list and keep purging
+            let (new_lru, new_size) = self.calculate_lru_list();
```
Comment (sakridge) [inline runtime/src/read_only_accounts_cache.rs:208]: What about `lru.clear()` here and then just recreate into it again?

File: runtime/src/read_only_accounts_cache.rs (hunk 1 above)
Comment (jeffwashington) [reply runtime/src/read_only_accounts_cache.rs:43]: done

File: runtime/src/read_only_accounts_cache.rs (hunk 2 above)
Comment (jeffwashington) [reply runtime/src/read_only_accounts_cache.rs]: done. bench tps shows a range of 1.3-1.8M entries in this list. With this new strategy, this lru list will keep growing and then remain at its largest size forever. Not a huge deal. I was imagining there were advantages to it using exactly what it needed each time the lru list is recalculated, which isn't too often. I can also add code to reallocate it smaller if we shrink size enough. I was trying to keep the code simple with less things to go wrong.

**Review by sakridge — APPROVED** [review]
lgtm

**Comment by jeffwashington** [comment]
perf tests:
mean_tps: 21412
max_tps: 62900
mean_confirmation_ms: 2511
max_confirmation_ms: 22199
99th_percentile_confirmation_ms: 9994
max_tower_distance: 164
last_tower_distance: 51.8333333333
slots_per_second: 2.127

**Pull request merged**